pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -439,9 +469,50 @@ func (c *Controller) HasSynced() bool {
 		!c.nodeInformer.HasSynced() {
 		return false
 	}
+
+	// after informer caches sync the first time, process resources in order
+	c.syncMu.Lock()
+	defer c.syncMu.Unlock()
+	if !c.initialSync {
+		if err := c.SyncAll(); err != nil {
+			log.Errorf("one or more errors force-syncing resources: %v", err)
+		}
+		c.initialSync = true
+	}
+
 	return true
 }
 
+func (c *Controller) SyncAll() error {
```

Comment:
It would be better if sync call would update caches for all the resources(node,service,endpoints, etc). And trigger one time push at last.  In this way we can prevent too many pushes.

Reply:
there won't be any connections until this is synced though so is there any overhead to the pushes?

Reply:
because we reject connection when not ready

Reply:
noticed that

Reply:
For multicluster, it can be different